datafusion/physical-plan/src/aggregates/mod.rs

```diff
@@ -369,14 +369,21 @@ impl AggregateExec {
         new_requirement.extend(req);
         new_requirement = collapse_lex_req(new_requirement);
 
-        let input_order_mode =
-            if indices.len() == groupby_exprs.len() && !indices.is_empty() {
-                InputOrderMode::Sorted
-            } else if !indices.is_empty() {
-                InputOrderMode::PartiallySorted(indices)
-            } else {
-                InputOrderMode::Linear
-            };
+        let indices: Vec<usize> = indices
```

Comment:
I recommend adding a comment here (perhaps copied from the description of the PR) explaining the subtelty involving grouping sets